cmd/clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -498,61 +500,156 @@ func (c *client) CreateMachines(machines []*clusterv1.Machine, namespace string)
 }
 
 func (c *client) DeleteClusters(namespace string) error {
```

Comment:
This function takes a namespace parameter to look and feel like a client-go function where `""` as a parameter means "all namespaces"?

I would prefer this to not be recursive though as this implementation can be a non terminating recursive call in the case when a cluster's namespace is empty. It may be unlikely but guaranteeing this function will return is better than nearly always terminating. 

Something like this maybe?
```
namespaces := []string{}
if namespaces == ""
//  get all namespaces from cluster and append to namespaces list
else
// append namespace passed in

for namespace in namespaces
// delete cluster collection in ns